### PR TITLE
Doc: update using serial port

### PR DESCRIPTION
**Pull request by fuzhongl**
Add PCIe serial port enabling part on NUC; which doesn't
have the internal serial console header.

Signed-off-by: fuzhongl <fuzhong.liu@intel.com>

### CODE REVIEW

**Comment by wenlingz** [comment]
@fuzhongl please refer make doc fail:

New errors/warnings found, please fix them:
==============================================

/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/develop.rst:84: WARNING: toctree contains reference to nonexisting document 'tutorials/enabling_serial_port'
/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/tutorials/using_serial_port.rst:60: WARNING: Figure caption must be a paragraph or empty comment.

.. figure:: images/NVMe-M.2-to-PCIe-adaptor.png
   :scale: 80

    NVMe M.2 to PCIe adaptor details
/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/tutorials/using_serial_port.rst:65: WARNING: Figure caption must be a paragraph or empty comment.

.. figure:: images/PCIe-serial-StarTech.png
   :scale: 80

    StarTech 2 Port Native PCI Express
/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/tutorials/using_serial_port.rst:70: WARNING: Figure caption must be a paragraph or empty comment.

.. figure:: images/PCIe-serial-LeKuo.png
   :scale: 80

    IO-PCE99100-2S

**Review by gvancuts — CHANGES_REQUESTED** [review]
There are many grammatical and punctuation issues here which I will leave to @dbkinder and @fitchbe to resolve.

It is premature also to remove the information pertaining to KBL NUC, let's keep it in place and add the one for TGL.

File: doc/tutorials/using_serial_port.rst [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
 <https://www.amazon.com/dp/B07BV1W6N8/ref=cm_sw_r_cp_ep_dp_wYm0BbABD5AK6>`_
 such a cable or you can build it yourself;
-refer to the `KBL Intel NUC product specification
-<https://www.intel.com/content/dam/support/us/en/documents/mini-pcs/nuc-kits/NUC7i7DN_TechProdSpec.pdf>`_
+refer to the `TGL Intel NUC product specification
+<https://ark.intel.com/content/www/us/en/ark/products/205594/intel-nuc-11-pro-kit-nuc11tnhi5.html>`_
 as shown below:
 
-.. figure:: images/KBL-serial-port-header.png
+.. figure:: images/NUC11TNH-serial-port-header.png
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:26]: This picture references the KBL NUC.

File: doc/tutorials/using_serial_port.rst [hunk 2]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+   multiboot2 /boot/acrn.bin  root=PARTUUID="b1bebafc-2b06-43e2-bf6a-323337daebc0“ uart=bdf@0x101
+
+Note:
+.. code-block:: bash
+
+   uart=bdf@0x100 for port 1
+
+   uart=bdf@0x101 for port 2
+
+   uart=bdf@0x101 is preferred for industry scenario; otherwise it can’t input in HV console after sos boots up.
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Fixed

File: doc/tutorials/using_serial_port.rst [hunk 3]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+
+Refer this `link
+<https://projectacrn.github.io/2.4/user-guides/hv-parameters.html>`_ to change bdf 01:00.1 to Hex format: 0x101;
+Then adding it into grub menu:
+
+.. code-block:: bash
+
+   multiboot2 /boot/acrn.bin  root=PARTUUID="b1bebafc-2b06-43e2-bf6a-323337daebc0“ uart=bdf@0x101
+
+Note:
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Fixed

File: doc/tutorials/using_serial_port.rst [hunk 4]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+
+.. code-block:: bash
+   
+   01:00.0 Serial controller: Asix Electronics Corporation Device 9100
+   01:00.1 Serial controller: Asix Electronics Corporation Device 9100
+
+Convert the bdf to Hex format
+*****************************
+
+Refer this `link
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Fixed.

File: doc/tutorials/using_serial_port.rst [hunk 5]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+
+    IO-PCE99100-2S
+
+Both of them are verified with this `commit
+<https://github.com/projectacrn/acrn-hypervisor/commit/9e838248c3ce4d6b68e1c5b068d10d566a06db10>`_.
+
+    
+Check the pdf information
+*************************
+Connection as following,boot into native and check the pdf information using: lspci
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst [hunk 6]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+.. figure:: images/PCIe-serial-LeKuo.png
+   :scale: 80
+
+    IO-PCE99100-2S
+
+Both of them are verified with this `commit
+<https://github.com/projectacrn/acrn-hypervisor/commit/9e838248c3ce4d6b68e1c5b068d10d566a06db10>`_.
+
+    
+Check the pdf information
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst:79]: Done!

File: doc/tutorials/using_serial_port.rst [hunk 7]
```diff
@@ -1,35 +1,38 @@
...
-.. figure:: images/KBL-serial-port-header.png
+.. figure:: images/NUC11TNH-serial-port-header.png
    :scale: 80
 
-   KBL serial port header details
+   NUC11TNH serial port header details
 
 
-.. figure:: images/KBL-serial-port-header-to-RS232-cable.jpg
+.. figure:: images/NUC11YNH-serial-port-header-to-RS232-cable.jpg
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Fix the typo.

File: doc/tutorials/using_serial_port.rst [hunk 8]
```diff
@@ -1,35 +1,38 @@
...
+Enable Serial Port on NUC
+#########################
 
+Option 1: Using the Serial Port on TGL Intel NUC
+================================================
+ 
 You can enable the serial console on the
-`KBL Intel NUC <https://www.amazon.com/Intel-Business-Mini-Technology-BLKNUC7i7DNH1E/dp/B07CCQ8V4R>`_
-(NUC7i7DNH). The KBL Intel NUC has a serial port header you can
+`TGL Intel NUC <https://item.jd.com/10026820852530.html>`_
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst:79]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Sorry for typo. Fixed it now.

File: doc/tutorials/using_serial_port.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst [hunk 9]
```diff
@@ -41,3 +44,69 @@ to connect to your host system.
...
+   01:00.1 Serial controller: Asix Electronics Corporation Device 9100
+
+Convert the bdf to Hex format
+*****************************
+
+Refer this `link
+<https://projectacrn.github.io/2.4/user-guides/hv-parameters.html>`_ to change bdf 01:00.1 to Hex format: 0x101;
+Then adding it into grub menu:
+
+.. code-block:: bash
```
[the remark this answers is not in the document]
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: changed to ..Note::

File: doc/tutorials/using_serial_port.rst [hunk 10]
```diff
@@ -1,35 +1,38 @@
-.. _connect_serial_port:
+.. _enabling_serial_port:
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:1]: The label you change here is used in another tutorial (`using_hybrid_mode_on_nuc.rst`).

I tried a `make doc` on  your PR but it still throws some errors. Have you validated that it works on your side?

**Comment by wenlingz** [comment]
@fuzhongl please refer:
New errors/warnings found, please fix them:
==============================================

/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/build/doc/rst/tutorials/using_serial_port.rst:17: WARNING: image file not readable: tutorials/images/NUC11TNH-serial-port.jpg

Makefile:65: recipe for target 'html' failed
make[1]: *** [html] Error 1
make[1]: Leaving directory '/home/acrn/workspace/make_doc/pme-yocto-build-make-doc-01/acrn-hypervisor/doc'
Makefile:129: recipe for target 'doc' failed
make: *** [doc] Error 2
+ error make doc
++ timestamp
++ date +%T
+ echo '08:39:44 ERROR: make doc'
08:39:44 ERROR: make doc

**Comment by wenlingz** [comment]
OK to verify doc

File: doc/tutorials/using_serial_port.rst (hunk 10 above)
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst]: Done!

File: doc/tutorials/using_serial_port.rst [hunk 11]
```diff
@@ -41,3 +44,64 @@ to connect to your host system.
...
+   :scale: 80
+
+   IO-PCE99100-2S
+
+    
+Check the BDF Information
+*************************
+Connection as following,boot into native and check the bdf information of the adapter using: ``lspci``
+
+.. figure:: images/PCIe-serial-Connection.png
```
Comment (fitchbe) [inline doc/tutorials/using_serial_port.rst:81]: This image has a typo ("yhe" instead of "the").

**Review by fitchbe — APPROVED** [review]
Pushed copy edits in using_serial_port.rst.

File: doc/tutorials/using_serial_port.rst (hunk 11 above)
Comment (fuzhongl) [reply doc/tutorials/using_serial_port.rst:80]: Done! Sorry for typo.

File: doc/tutorials/using_serial_port.rst [hunk 12]
```diff
@@ -41,3 +44,64 @@ to connect to your host system.
...
+
+.. figure:: images/PCIe-serial-LeKuo.png
+   :scale: 80
+
+   IO-PCE99100-2S
+
+    
+Check the BDF Information
+*************************
+Connection as following,boot into native and check the bdf information of the adapter using: ``lspci``
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:78]:
```suggestion
Connection as following, boot into native and check the BDF information of the adapter using: ``lspci``
```

@fitchbe anything else to add/change to this?

File: doc/tutorials/using_serial_port.rst [hunk 13]
```diff
@@ -41,3 +44,64 @@ to connect to your host system.
...
+   01:00.0 Serial controller: Asix Electronics Corporation Device 9100
+   01:00.1 Serial controller: Asix Electronics Corporation Device 9100
+
+Convert the BDF to Hex Format
+*****************************
+
+Refer this :ref:`hv-parameters` to change bdf 01:00.1 to Hex format: 0x101;
+Then adding it into grub menu:
+
+.. Note::
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:96]: I did not mean to change that into a note, I was questioning the use of `bash` as the `code-block` style.

File: doc/tutorials/using_serial_port.rst [hunk 14]
```diff
@@ -41,3 +44,64 @@ to connect to your host system.
...
+   multiboot2 /boot/acrn.bin  root=PARTUUID="b1bebafc-2b06-43e2-bf6a-323337daebc0“ uart=bdf@0x101
+
+.. Note::
+
+   uart=bdf@0x100 for port 1
+
+   uart=bdf@0x101 for port 2
+
+   uart=bdf@0x101 is preferred for industry scenario; otherwise it can’t input in Hypervisor console after Service VM boots up.
+   There is no this limitation for hybrid and hybrid_rt scenario.
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:107]: @fitchbe it sounds like we could improving the phrasing a bit on this one. Will you do this after it has been merged?

File: doc/tutorials/using_serial_port.rst [hunk 15]
```diff
@@ -41,3 +44,64 @@ to connect to your host system.
 
 Note that If you want to use the RS232 DB9 female/female cable, choose
 the **cross-over** type rather than **straight-through** type.
+
+Option 2: Enabling PCIe Serial Port on TGL Intel NUC
+====================================================
+
+If there is no internal serial port header on your NUC; you can enable PCIe serial port to debug the issues effeciently.
```
Comment (gvancuts) [inline doc/tutorials/using_serial_port.rst:51]:
```suggestion
If there is no internal serial port header on your NUC; you can enable PCIe serial port to debug the issues efficiently.
```

**Review by gvancuts — APPROVED** [review]
Thanks. @fitchbe will you take another pass at this later?

File: doc/tutorials/using_serial_port.rst (hunk 14 above)
Comment (fitchbe) [reply doc/tutorials/using_serial_port.rst:107]: Not sure what happened; in my local cop of the branch, I fixed these several spots that you've highlighted but don't see the fixes here. Probably my lack of Git experience. I will do another complete pass on this file after merge.

**Pull request merged**